packages/teams-js/src/public/video.ts

```diff
@@ -88,11 +88,33 @@ export namespace video {
     notifyError: (errorMessage: string) => void,
   ) => void;
 
+  /**
+   * Predefined failure reasons for preparing the selected video effect
+   * @beta
+   */
+  export enum EffectFailureReason {
+    /**
+     * A wrong effect id is provided
+     */
+    InvalidEffectId = 'InvalidEffectId',
+    /**
+     * The effect can't be initialezed
+     */
+    InitializationFailure = 'InitializationFailure',
+  }
+
+  enum EffectSuccessReason {
+    EffectPrepared = 'EffectPrepared',
+
+    NoPromiseReturned = 'NoPromiseReturned',
+  }
+
   /**
    * Video effect change call back function definition
+   * Return a Promise which will be resolved when the effect is prepared, or reject with a failure reason
    * @beta
    */
-  export type VideoEffectCallBack = (effectId: string | undefined) => void;
+  export type VideoEffectCallBack = (effectId: string | undefined) => Promise<void> | void;
```

Comment:
this is also used in video.registerForVideoFrame(...). Am I correct in assuming that for that function, the callback should be called every time a frame is ready?

I don't think I understand why this is returning either void or Promise<void>. Shouldn't it just return a Promise in both cases? It will be very confusing for developers trying to figure out whether they should return a promise or not.

Reply:
In your design doc you have this set to just return a Promise. I don't think there's any need to maintain backwards compatibility here, this capability and it's functions are already marked as @beta.

Reply:
Updated